test/tools/jsontests/StateTests.cpp

```diff
@@ -49,7 +49,7 @@ void doStateTests(json_spirit::mValue& _v, bool _fillin)
 		string testname = i.first;
 		json_spirit::mObject& o = i.second.get_obj();
 
-		if (_fillin)
+		if (_fillin && !Options::get().createRandomTest)
```

Comment:
what? why? 
if it is a filler then it is being checked that testname in json object match the test filename. 
unless it is a filling of a randomstate test that does not have a file.